example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java

```diff
@@ -0,0 +1,133 @@
+        }
+
+        int[] hashMap = new int[10];
+
+        try
+        {
+            initializeVariables();
+
+            for(int i = 0; i < numberOfImages; i++) {
+
+                if(i % 100 == 0) {System.out.println("Number of images extracted: " + i);}
+
+                for(int p = 0; p < numberOfPixels; p++) {
+                    int gray = 255 - inImage.read();
+                    imgPixels[p] = 0xFF000000 | (gray<<16) | (gray<<8) | gray;
+                }
+
+                image.setRGB(0, 0, numberOfColumns, numberOfRows, imgPixels, 0, numberOfColumns);
+
+                int label = inLabel.read();
+
+                hashMap[label]++;
+                File outputfile = new File(outputPath + "/" + label + "/" + hashMap[label] + ".png");
+
+                ImageIO.write(image, "png", outputfile);
+            }
+
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            closeFileStreams();
+        }
+    }
+
+    private void closeFileStreams() {
+        if (inImage != null) {
+            try {
+                inImage.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        if (inLabel != null) {
+            try {
+                inLabel.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private void initializeVariables() {
+        try {
+
+            inImage = new FileInputStream(inputImagePath);
+            inLabel = new FileInputStream(inputLabelPath);
+
+            magicNumberImages = (inImage.read() << 24) | (inImage.read() << 16) | (inImage.read() << 8) | (inImage.read());
+            numberOfImages = (inImage.read() << 24) | (inImage.read() << 16) | (inImage.read() << 8) | (inImage.read());
+            numberOfRows = (inImage.read() << 24) | (inImage.read() << 16) | (inImage.read() << 8) | (inImage.read());
+            numberOfColumns = (inImage.read() << 24) | (inImage.read() << 16) | (inImage.read() << 8) | (inImage.read());
+
+            magicNumberLabels = (inLabel.read() << 24) | (inLabel.read() << 16) | (inLabel.read() << 8) | (inLabel.read());
+            numberOfLabels = (inLabel.read() << 24) | (inLabel.read() << 16) | (inLabel.read() << 8) | (inLabel.read());
+
+            image = new BufferedImage(numberOfColumns, numberOfRows, BufferedImage.TYPE_INT_ARGB);
+            numberOfPixels = numberOfRows * numberOfColumns;
+            imgPixels = new int[numberOfPixels];
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void createOutputDirectories(String outputPath)
+    {
+        File outputDir = new File(outputPath);
+        outputDir.mkdir();
+        for(int ix = 0; ix < 10; ++ix)
+        {
+            new File(outputPath + "" + ix).mkdir();
```

Comment:
`+ ""` isn't necessary here

Reply:
It wasn't allowing me to add them otherwise